calendar-bundle/src/Resources/contao/dca/tl_calendar_events.php

```diff
@@ -63,7 +63,7 @@
 			(
 				'id' => 'primary',
 				'alias' => 'index',
-				'pid,start,stop,published' => 'index'
+				'pid,published,featured,start,stop' => 'index'
```

Comment:
I'd leave out `featured` so that `start,stop` can at least be used (once) in most queries as last filter step. As discussed [here](https://github.com/contao/contao/pull/2234#discussion_r483508825): If someone has millions of calendar entries that would be shown in a certain view they would be better of using a dedicated `featured` index.

Reply:
On a side note: I'm not sure about the current implementation but esp. with sorting needed it's sometimes better to have two separate queries: `SELECT id FROM table WHERE <conditions using a, b> ORDER BY start` + `SELECT * FROM table WHERE id IN <ids>` and indexes that are in the (reverse) order of the order field: `start, a, b` or `start` /  `a,b`.

Reply:
There are only two cases in the front end: listing all events or showing a single event. All listing modules (event list and calendar) do check the `featured` column, therefore it makes sense to have it in the index. And the event reader uses the index on the `alias` column, so we no not have the `pid_published_start_stop` case anywhere.